sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -223,10 +221,14 @@ CosmosKeyCredential cosmosKeyCredential() {
         }
         Database wrappedDatabase = new Database();
         wrappedDatabase.setId(databaseSettings.getId());
-        return asyncDocumentClient.createDatabase(wrappedDatabase, ModelBridgeInternal.toRequestOptions(options))
-                   .map(databaseResourceResponse -> ModelBridgeInternal.createCosmosAsyncDatabaseResponse(databaseResourceResponse,
-                       this))
-                   .single();
+        final CosmosDatabaseRequestOptions requestOptions = options;
+        return withContext(context -> createDatabase(wrappedDatabase, requestOptions, context)).subscriberContext(reactorContext -> {
```

Comment:
As per conversation with Sameekhsa and Anu, core team will provide withContext, fluxContext api with map attributes to to handle nested calls on sdk more cleanly

Reply:
This should already be available to use on latest core i.e ` <version>1.4.0-beta.1</version> `

Reply:
Checked offline with Sameeksha 1.4.0-beta.1 does not have the requested overload

Reply:
Discussed with Sameeksha offline , as of now cosmos sdk wont need it